gctscript/modules/ta/indicators/indicators_test.go

```diff
@@ -0,0 +1,571 @@
+	t.Parallel()
+	_, err := ema()
+	if err != nil {
+		if !errors.Is(err, objects.ErrWrongNumArguments) {
+			t.Error(err)
+		}
+	}
+
+	v := &objects.String{Value: testString}
+	_, err = ema(ohlcvData, v)
+	if err != nil {
+		if err.Error() != errFailedConversion {
+			t.Error(err)
+		}
+	}
+
+	_, err = ema(ohlcvData, &objects.Int{Value: 14})
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = ema(ohlcvDataInvalid, &objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	_, err = ema(&objects.String{Value: testString}, &objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	validator.IsTestExecution.Store(true)
+	ret, err := ema(ohlcvData, &objects.Int{Value: 14})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if (ret == &objects.Array{}) {
+		t.Error("expected empty Array on test execution received data")
+	}
+	validator.IsTestExecution.Store(false)
+}
+
+func TestSMA(t *testing.T) {
+	t.Parallel()
+	_, err := sma()
+	if err != nil {
+		if !errors.Is(err, objects.ErrWrongNumArguments) {
+			t.Error(err)
+		}
+	}
+
+	v := &objects.String{Value: testString}
+	_, err = sma(ohlcvData, v)
+	if err != nil {
+		if err.Error() != errFailedConversion {
+			t.Error(err)
+		}
+	}
+
+	_, err = sma(ohlcvData, &objects.Int{Value: 14})
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = sma(ohlcvDataInvalid, &objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	_, err = sma(&objects.String{Value: testString}, &objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	validator.IsTestExecution.Store(true)
+	ret, err := sma(ohlcvData, &objects.Int{Value: 14})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if (ret == &objects.Array{}) {
+		t.Error("expected empty Array on test execution received data")
+	}
+	validator.IsTestExecution.Store(false)
+}
+
+func TestMACD(t *testing.T) {
+	t.Parallel()
+	_, err := macd()
+	if err != nil {
+		if !errors.Is(err, objects.ErrWrongNumArguments) {
+			t.Error(err)
+		}
+	}
+
+	v := &objects.String{Value: testString}
+	_, err = macd(ohlcvData, &objects.Int{Value: 12}, &objects.Int{Value: 26}, v)
+	if err != nil {
+		if err.Error() != errFailedConversion {
+			t.Error(err)
+		}
+	}
+
+	_, err = macd(ohlcvData, &objects.Int{Value: 12}, &objects.Int{Value: 26}, &objects.Int{Value: 9})
+	if err != nil {
+		t.Error(err)
+	}
+
+	_, err = macd(ohlcvDataInvalid,
+		&objects.String{Value: testString},
+		&objects.String{Value: testString},
+		&objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	_, err = macd(&objects.String{Value: testString},
+		&objects.String{Value: testString},
+		&objects.String{Value: testString},
+		&objects.String{Value: testString})
+	if err == nil {
+		t.Error("expected conversion failed error")
+	}
+
+	validator.IsTestExecution.Store(true)
```

Comment:
These tests are always passing for me, but I find this part weird. You're setting the tests to parallel but changing atomic values. Golang doesn't wait for these to change, right? So in theory you could get to these points in the code and have them be false when it executes, right?